### PR TITLE
CI: Add GPU Vs CPU dry HS comparison job

### LINKED ISSUE #1926: Dry HS: compare CPU vs GPU results
Similar to #1864 , but with the dry HS test case

### PR DESCRIPTION
**Pull request by valeriabarra**
## Purpose 
The purpose of this PR is to add a CI job for the GPU Vs CPU comparison for the dry HS test case.

Closes #1926 

## Content
- Renamed the existing GPU and comparison job for dry BW
- Added a GPU and comparison job for the dry HS test case

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by valeriabarra** [comment]
bors try

**Comment by valeriabarra** [comment]
The plots look identical: 

- GPU dry HS high-top: 
    - `rho`, day 8, level 10:
rho_day8_level10


   - `rho_e_tot`, day 8, level 10:
rho_e_tot_day8_level10


   - `u`, day 8, level 10:
u_day8_level10


   - `w`, day 8, level 10+half:
w_day8_level10


- CPU dry HS high-top: 
    - `rho`, day 8, level 10:
rho_day8_level10


   - `rho_e_tot`, day 8, level 10:
rho_e_tot_day8_level10



   - `u`, day 8, level 10:
u_day8_level10



   - `w`, day 8, level 10+half:
w_day8_level10

**Comment by valeriabarra** [comment]
@charleskawczynski I'm not sure why CI failed. I only slightly renamed the existing dry BW job and added the new HS comparison job. 

I can run the comparison just fine locally on my machine and it looks similar to the dry BW test case you added.

**Comment by charleskawczynski** [comment]
> @charleskawczynski I'm not sure why CI failed. I only slightly renamed the existing dry BW job and added the new HS comparison job.

It timed out, waiting for gpu resources.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -805,20 +805,35 @@ steps:
   - group: "GPU"
     steps:
 
-      - label: "GPU: GPU implicit baro wave"
+      - label: "GPU: GPU implicit dry baro wave"
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:808]: Can we just leave out the "dry" for the dry case, and add moist for the moist case? I think the naming scheme is easier to follow if we leave out default values (dry is the default)

Comment (valeriabarra) [reply .buildkite/pipeline.yml:808]: Sure we can! I will change it

**Review by charleskawczynski — APPROVED** [review]

**Comment by valeriabarra** [comment]
@charleskawczynski the time-out on the GPU comparison jobs happened again. This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?

**Comment by charleskawczynski** [comment]
> This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?

It might be easier to review if this PR only added new jobs, and didn't also modify existing ones.

**Comment by valeriabarra** [comment]
> > This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?
> 
> It might be easier to review if this PR only added new jobs, and didn't also modify existing ones.

Okay, it was just a simple renaming (and I had made sure that I changed job id's and artifact paths accordingly). But I reverted it for simplicity.

**Comment by charleskawczynski** [comment]
No problem, this looks fine, except we need to add the `sphere_held_suarez_rhoe_hightop` key for that job (it currently only has a label). This is why it previously hung

**Pull request merged**